File: src/main/java/club/someoneice/thursday_delight/ThursdayMain.java
package club.someoneice.thursday_delight;

import com.mojang.logging.LogUtils;

import net.minecraft.client.renderer.ItemBlockRenderTypes;
import net.minecraft.client.renderer.RenderType;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
import net.minecraftforge.event.server.ServerStartingEvent;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

import org.slf4j.Logger;

import javax.annotation.Nonnull;


@Mod(ThursdayMain.MODID)
public class ThursdayMain {
    public static final String MODID = "thursday_delight";
    private static final Logger LOGGER = LogUtils.getLogger();

    public static final CreativeModeTab TAB = new CreativeModeTab("thursday_group") {
        @Nonnull @Override
        public ItemStack makeIcon() {
            return new ItemStack(Items.COOKED_CHICKEN);
        }
    };

    public ThursdayMain() {
        LOGGER.info("Thursday Delight loaded now");
        IEventBus modEventBus = FMLJavaModLoadingContext.get().getModEventBus();
        modEventBus.addListener(this::clientSetup);

        ItemList.ITEMS.register(modEventBus);
        BlockList.BLOCKS.register(modEventBus);
        BlockEntityList.BLOCKENTITY.register(modEventBus);

        MinecraftForge.EVENT_BUS.register(this);
    }

    public void clientSetup(final FMLClientSetupEvent event) {
        ItemBlockRenderTypes.setRenderLayer(BlockList.ChiliCrop.get(), RenderType.cutout());
    }

    @SubscribeEvent
    public void onServerStarting(ServerStartingEvent event) {
    }

}

File: src/main/kotlin/club/someoneice/thursday_delight/ThursdayMain.kt
package club.someoneice.thursday_delight

// import club.someoneice.thursday_delight.client.Screens

import net.fabricmc.fabric.api.client.itemgroup.FabricItemGroupBuilder
import net.minecraft.item.ItemGroup
import net.minecraft.item.ItemStack
import net.minecraft.item.Items
import net.minecraft.util.Identifier


object ThursdayMain {
    const val modid: String = "thursday_delight"
    val thursdayGroup: ItemGroup = FabricItemGroupBuilder.build(Identifier(modid, "thursday_group")) {
        ItemStack(Items.COOKED_CHICKEN)
    }
}

@Suppress("unused")
fun init() {
    println("Now install Thursday Delight!")
    RegisterInit.init()
    RegistryEntityHelper.init()

    // Why I use it?
    // Screens.init()
}

